.github/workflows/sandbox-rebuild.yml

```diff
@@ -0,0 +1,46 @@
+name: SandBox Rebuild
+
+on: push
+
+env:
+  AWS_REGION: ${{ vars.AWS_REGION }}
+  BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+
+permissions:
+  id-token: write # This is required for requesting the JWT
+  contents: read  # This is required for actions/checkout
+
+jobs:
+  AssumeRoleAndStartExecution:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
```

Comment:
The first job should be to check if the current push to the main branch and if it’s the case, then we should stop to run this action, because we don’t need to have sandbox rebuild for the main branch